src/include/simeng/arch/aarch64/Architecture.hh

```diff
@@ -106,6 +119,9 @@ class Architecture : public arch::Architecture {
   /** Modulo component used to define the frequency at which the VCT is updated.
    */
   double vctModulo_;
+
+  /** A copy of the value of the SVCR system register. */
+  static uint64_t SVCRval_;
```

Comment:
Unsure of why there needs to be a copy of this register? 

The current implementation only sets and gets this within instruction_execute.cc (bar one get in instruction_address.cc). I am unsure of the intended functionality but it seems like this may cause problems with out of order execution. This value can be set on one cycle and the new value immediately read on the next cycle, possibly by an instruction earlier in program order e.g. causing a RAW hazard. Should a system register be set within the writeback stage to avoid hazards? If there is an issue here then it won't be caught by the tests as they run in emulation mode only

Reply:
After looking at changes to the instruction_decode.cc, why is this register not handled in the usual way e.g. setting it as a normal source/destination register? See https://github.com/UoB-HPC/SimEng/blob/51f045ddb69dde9b8f945487bd82c39ebd80176b/src/lib/arch/aarch64/Instruction_decode.cc#L343

Reply:
The new implementation looks much better and I think is easier to understand. For tests I was looking at sme.cc, but Exception.cc has relevant tests and runs in all simulation modes. Thanks for the in depth explanation